### PR TITLE
move test_util out of q, rework due to es6

### PR DESCRIPTION
**Pull request by CrispusDH**
- [x] rework `test_util`
- [x] check passed tests locally
- [x] check correct behavior if some test failed
- [x] update node version in CircleCI to LTS (tests were failed on previous v8)
- [ ] change octaral literals to something different

I don't know how to fix it:
```
Octal literals are not allowed in strict mode
console.log('\n>>> \033[1;32mpass\033[0m');
```

### CODE REVIEW

**Comment by CrispusDH** [comment]
@cnishina , could you take a look at this PR, please

**Comment by cnishina** [comment]
Let's leave the octal literals as is. This should be blocked by #5038. In #5038, the PR turns on more of the error tests.

**Comment by CrispusDH** [comment]
Reference #4995

File: scripts/test/test_util.js [hunk 1]
```diff
@@ -194,35 +192,35 @@ var CommandlineTest = function(command) {
...
-          failed = true;
-          console.log('\n>>> \033[1;31mfail: ' + err.toString() + '\033[0m');
-        }).fin(function() {
-          runTests(i + 1);
-        }).done();
-      } else {
-        console.log('Summary: ' + (failed ? 'fail' : 'pass'));
-        process.exit(failed ? 1 : 0);
+        await test.run();
+        console.log('\n>>> \033[1;32mpass\033[0m');
```
Comment (cnishina) [inline scripts/test/test_util.js]:
```suggestion
        console.log('\n>>> \\033[1;32mpass\\033[0m');
```
Per https://stackoverflow.com/questions/36878850/octal-literals-are-not-allowed-in-strict-mode

File: scripts/test/test_util.js [hunk 2]
```diff
@@ -194,35 +192,35 @@ var CommandlineTest = function(command) {
...
+      } catch (error) {
+        failed = true;
+        console.log('\n>>> \033[1;31mfail: ' + err.toString() + '\033[0m');
```
Comment (cnishina) [inline scripts/test/test_util.js]:
```suggestion
        console.log('\n>>> \\033[1;31mfail: ' + err.toString() + '\\033[0m');
```
Per https://stackoverflow.com/questions/36878850/octal-literals-are-not-allowed-in-strict-mode

**Comment by CrispusDH** [comment]
It fix change `octal` to plain text. So, in console is just `>>> \033[1;32mpass\033[0m`

**Comment by CrispusDH** [comment]
I suggest change:
1. `\n>>> \033[1;31mfail: ' + err.toString() + '\033[0m`, that produce just "pass" to `\n>>> test was passed`
2. `\n>>> \033[1;31mfail: ' + err.toString() + '\033[0m`, change to `\n>>> test was failed. Error: ${error}`

**Comment by cnishina** [comment]
Hmmm... something is broken. The tests pass but nothing is running.

**Comment by cnishina** [comment]
I'm going to work on a separate PR and I'll get it to work today.

**Comment by CrispusDH** [comment]
Wow, CircleCI passed! We need just fix octal message. @cnishina

**Comment by CrispusDH** [comment]
@cnishina , sorry but I'm not aware how to fix it if you want to show test duration :)

**Comment by cnishina** [comment]
So here's how I would fix up the PR.... https://github.com/angular/protractor/pull/5058

The octal is problematic but we are not required to update everything. Leaving that last section as is in test_util is fine. The bigger concern is to remove q. Please feel free to borrow / steal as much as you need.

**Pull request merged**